### PR TITLE
App start: native spans

### PR DESCRIPTION
**Pull request by buenaflor**
## :scroll: Description

Adds native spans to app start transaction

```[tasklist]
### Blocked by
- [ ] https://github.com/getsentry/sentry-dart/pull/2009
``` 

## :bulb: Motivation and Context

Improve app start transaction and give insight into the native layer times

## :green_heart: How did you test it?
Unit, Manual with spotlight

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: flutter/lib/src/integrations/native_app_start_integration.dart [hunk 1]
```diff
@@ -101,12 +106,37 @@ class NativeAppStartIntegration extends Integration<SentryFlutterOptions> {
...
+            // ignore: invalid_use_of_internal_member
+            _hub.options.logger(
+                SentryLevel.warning, 'Failed to parse native span times: $e');
+            continue;
+          }
+        }
+
+        // We want to sort because the native spans are not guaranteed to be in order.
+        // Performance wise this won't affect us since the native span amount is very low.
+        nativeSpanTimes.sort((a, b) => a.start.compareTo(b.start));
```
Comment (buenaflor) [inline flutter/lib/src/integrations/native_app_start_integration.dart]: also happy to hear other recommendations to solve this if this is not sufficient enough

File: flutter/lib/src/sentry_flutter.dart [hunk 2]
```diff
@@ -48,6 +48,9 @@ mixin SentryFlutter {
   }) async {
     final flutterOptions = SentryFlutterOptions();
 
+    // ignore: invalid_use_of_internal_member
+    mainIsolateStartTime = flutterOptions.clock();
```
Comment (stefanosiano) [inline flutter/lib/src/sentry_flutter.dart]: this should be removed, same reaason of the other PR (not changing the mainIsolateStartTime after being initialized)

File: flutter/test/integrations/native_app_start_integration_test.dart [hunk 3]
```diff
@@ -207,7 +263,7 @@ void main() {
       final processor = fixture.options.eventProcessors.first;
       tracer = fixture.createTracer();
       final transaction = SentryTransaction(tracer);
-      final enriched =
+      enriched =
```
Comment (stefanosiano) [inline flutter/test/integrations/native_app_start_integration_test.dart:265]: is the removal of `final` an overlook?

**Review by stefanosiano — APPROVED** [review]
just a detail, but good for the rest!

File: flutter/lib/src/integrations/native_app_start_integration.dart (hunk 1 above)
Comment (stefanosiano) [reply flutter/lib/src/integrations/native_app_start_integration.dart]: guess it should be enough

File: flutter/test/integrations/native_app_start_integration_test.dart (hunk 3 above)
Comment (buenaflor) [reply flutter/test/integrations/native_app_start_integration_test.dart:265]: I set `late SentryTransaction enriched;` so I can access it in each test

**Comment by buenaflor** [comment]
@stefanosiano I've updated the span description to `Before Sentry Init Setup `

File: flutter/lib/src/sentry_flutter.dart [hunk 4]
```diff
@@ -48,6 +48,9 @@ mixin SentryFlutter {
   }) async {
     final flutterOptions = SentryFlutterOptions();
 
+    // ignore: invalid_use_of_internal_member
+    sentrySetupStartTime = flutterOptions.clock();
```
Comment (stefanosiano) [inline flutter/lib/src/sentry_flutter.dart]:
```suggestion
    sentrySetupStartTime ??= flutterOptions.clock();
```
so if the user initializes the SDK multiple times it's not an issue

File: flutter/lib/src/sentry_flutter.dart (hunk 2 above)
Comment (stefanosiano) [reply flutter/lib/src/sentry_flutter.dart]: just leaving a comment for future readers. Dart static variables are lazy, initialized only when accessed, so we cannot rely on static variables.

**Comment by stefanosiano** [comment]
ok, last thing and can be merged 👍

**Pull request merged**